src/store/reducers/dropdown.ts

```diff
@@ -0,0 +1,44 @@
+import { Reducer } from "redux";
+
+import { List } from "immutable";
+import { DropdownAction, DropdownActionsType } from "../actions/dropdown";
+
+export enum DropdownType {
+  INFO = "info",
+  WARN = "warn",
+  ERROR = "error",
+  SUCCESS = "success"
+  // CUSTOM is not currently supported.
+}
+
+export interface DropdownMessage {
+  type: DropdownType;
+  id: string; // autogenerated
+  title: string;
+  message: string;
+}
+
+export type DropdownState = {
+  messages: List<DropdownMessage>;
+};
+
+export const reducer: Reducer<DropdownState> = (
+  prevState = {
+    messages: List()
+  },
+  untypedAction
+) => {
+  const action = untypedAction as DropdownAction;
+  switch (action.type) {
+    case DropdownActionsType.ADD_DROPDOWN_MESSAGE:
```

Comment:
Like I mentioned, we could alternatively do this by listening for other redux actions that directly correspond to stuff happening, but this approach is fine too. Logging for posterity